scrapy/crawler.py

```diff
@@ -149,9 +149,9 @@ def start(self, stop_after_crawl=True):
                 return
             d.addBoth(lambda _: self._stop_reactor())
 
-        if self.settings.getbool('DNSCACHE_ENABLED'):
-            reactor.installResolver(CachingThreadedResolver(reactor))
-
+        cache_size = self.settings.getint('DNSCACHE_SIZE') if self.settings.getbool('DNSCACHE_ENABLED') else 0
+        reactor.installResolver(CachingThreadedResolver(reactor, cache_size,
+                                                            self.settings.getint('DNS_TIMEOUT')))
```

Comment:
No. And the usefulness for a timeout less than a second is questionable.

The interface in theory supports a list of values `(1, 3, 11, 45)`, which contains staggerable timeouts, though. But it is only `sum()`med in the default implementation.

To quote [the docs](https://twistedmatrix.com/documents/14.0.1/api/twisted.internet.interfaces.IResolverSimple.html#getHostByName): "timeout:  Number of seconds after which to reissue the query. When the last timeout expires, the query is considered failed. (type: Sequence of int)"